developer/tutorials/Architecture of nopCommerce.md

```diff
@@ -0,0 +1,71 @@
+
+Which means projects can only have dependency to the other projects which resides inward from the current project. For example if you see the above diagram `Nop.Data` project can depend on Nop.Core project and can have `Nop.Core` and dependency, but nop.Core cannot depend on Nop.Data similarly Nop.Services can have Nop.Data and Nop.Core and it's dependency but neither Nop.Core nor `Nop.Data` can have Nop.Service as their dependency. Which means projects can only have other project as dependency only if it resides inward or more center from the layer current project resides. Which is the key for code decoupling. The main feature/advantage of this approach/architecture is that, now if we want to test the Application Core then we can do so even if we do not have any `UI` for our application , Because the Application core does not depends on `UI layer`. Or we can change our `UI framework` from `Razor` view engine and `JQuery` to `Angular` or `React` or `Vue` without effecting our Application Core and can use same core to build Mobile Application or Desktop Application. And all of this without changing a bit of code in our Application Core.
+
+## Explain projects in nopCommerce Solution
+
+### Application Core
+
+This is the innermost layer in the nopCommerce architecture system. It is the core of the application. All the data access logic and Business class resides inside this layer. In nopCommerce solution we can find all of the projects for this layer inside "Libraries" directory. According to nopCommerce architecture this layer contains further three layers/projects.
+
+#### Nop.Core
+
+This project contains a set of core classes for nopCommerce. This project is at the center of the architecture. This project does not have any dependencies to other projects in the solution. This project contains core classes which are shared to entire solution like Domain Entities, Cashing, Events and other Helper classes.
+
+#### Nop.Data
+
+This project lives right after the `Nop.Core` project. So, this project has dependency to project that lives towards the middle from this project in the architecture that is `Nop.Core` project. Beside `Nop.Core` project it does not have any dependency to any other project in the solution. The `Nop.Data` project contains a set of classes and functions for reading from and writing to a database or other data store. It helps separate data-access logic from your business objects. nopCommerce uses the `Entity Framework Core` Code-First approach. It allows you to define entities in the source code (all core entities are defined into Nop.Core project), and then get `Entity Framework Core` to generate the database from that. That's why it's called Code-First. You can then query your objects using `LINQ`, which gets translated to SQL behind the scenes and executed against the database. nopCommerces use `Fluent API` to fully customize the persistence mapping.
```

Comment:
In the latest version (4.30), we abandoned the use of EF Core by replacing the implementation with Linq2db in conjunction with the Fluent Migrator. Please change this section of the article to reflect current changes in version 4.30